pkg/build/builder/docker.go

```diff
@@ -264,7 +264,10 @@ func (d *DockerBuilder) buildInfo() []dockerfile.KeyValue {
 // consume.
 func (d *DockerBuilder) buildLabels(dir string) []dockerfile.KeyValue {
 	labels := map[string]string{}
-	sourceInfo := d.git.GetInfo(dir)
+	sourceInfo := &s2iapi.SourceInfo{}
+	if d.build.Spec.Source.Git != nil {
```

Comment:
ultimately we want to move away from checking type fields and just go with "if it's non-nil, it's intended to be used". ( @smarterclayton has convinced me that's a better approach.  it's particularly useful where it might make sense to supply 2 or more types, like gitsource+a separate dockerfile)